client-csharp/tests/SessioonPoolTest.cs

```diff
@@ -0,0 +1,921 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+using System.Collections.Generic;
+using System;
+using iotdb_client_csharp.client.utils;
+using System.Threading.Tasks;
+
+namespace iotdb_client_csharp.client.test
+{
+    public class SessionPoolTest
+    {
+        public string host = "localhost";
+        public int port = 6667;
+        public string user = "root";
+        public string passwd = "root";
+        public int fetch_size = 10000;
+        public int processed_size = 8;
+        public bool debug = false;
+        int pool_size = 3;
+
+        public void Test(){
+            Task task;
+            
+            task = TestInsertRecord();
+            task.Wait();
+            
+            task = TestCreateMultiTimeSeries();
+            task.Wait();
+            task = TestGetTimeZone();
+            task.Wait();
+            task = TestInsertStrRecord();
+            task.Wait();
+            task = TestInsertRecords();
+            task.Wait();
+            task = TestInsertRecordsOfOneDevice();
+            task.Wait();
+            task = TestInsertTablet();
+            task.Wait();
+            task = TestInsertTablets();
+            task.Wait();          
+            task = TestSetAndDeleteStorageGroup();
+            task.Wait();
+            task = TestCreateTimeSeries();
+            task.Wait();
+            task = TestDeleteStorageGroups();
+            task.Wait();
+            task = TestCheckTimeSeriesExists();
+            task.Wait();
+            task = TestSetTimeZone();
+            task.Wait();
+            task = TestDeleteData();
+            task.Wait();
+            task = TestNonSql();
+            task.Wait();
+            task = TestSqlQuery();
+            task.Wait();
+            
+        }
+
+        public async Task TestInsertRecord(){
+            var session_pool = new SessionPool(host, port, pool_size);
+            int status;
+            await session_pool.open(false);
+            if(debug){
+                session_pool.open_debug_mode();
+            }
+            System.Diagnostics.Debug.Assert(session_pool.is_open());
+            status = await session_pool.delete_storage_group_async("root.97209_TEST_CSHARP_CLIENT_GROUP");
+
+            status = await session_pool.create_time_series_async("root.97209_TEST_CSHARP_CLIENT_GROUP.TEST_CSHARP_CLIENT_DEVICE.TEST_CSHARP_CLIENT_TS1", TSDataType.TEXT, TSEncoding.PLAIN, Compressor.UNCOMPRESSED);
+          
+            System.Diagnostics.Debug.Assert(status == 0);
```

Comment:
For such assertions, special libs as Xunit/Nunit can be used.

Reply:
good idea, we directly used system.Diagnostics because we want less dependency